### DISCUSSION TITLE
native mothod could be traced by btrace?

### DISCUSSION THREAD

**Discussion by zhangchengzhangtuo** [btraceio/btrace#626] [Q&A]
when I  use btrace to trace java.lang.String#intern, I found it did not work, so native method could not be traced by btrace? everyone who knows

**Comment by jbachorik** [comment]
This is not currently possible, unfortunately.
I have already prototyped support for tracing native methods but that would work only when the agent is started with the application and not for the dynamic attach. In short, it would use the JVM provided functionality to add a suffix to all the native method names and then generate plain Java methods with the original names and parameters, where the instrumentation could be injected. But since we can not add/remove methods from classes during class retransformation this would never work for the dynamic attach.
